### PR TITLE
fix(remoteconfig): fix remoteconfig when flare data in payload

### PR DESCRIPTION
**Pull request by ZStriker19**
Remote config was broken entirely due to the addition of tracer flare data in the config sent from the agent. Example incoming RC config:
```
data = {'metadata': [{'id': 'configuration_order', 'product_name': 'AGENT_CONFIG', 'sha256_hash': 'ddfc2c7b5ee1710aa915edfccd8a0d452784d946cebae0554485b5c0539a9e2c', 'length': 198, 'tuf_version': 2, 'apply_state': 2, 'apply_error': None}, {'id': 'f6c80fdcc00b702c54ff6ae5ff2ac7f16d9afef109bdf53ee990376455301ab2', 'product_name': 'APM_TRACING', 'sha256_hash': '098cb5a0d27fce648cdd4c6e686038282b64ffee7f42b7238a78552c91948d11', 'length': 616, 'tuf_version': 3, 'apply_state': 2, 'apply_error': None}], 'config': [{'internal_order': ['flare-log-level.trace', 'flare-log-level.debug', 'flare-log-level.info', 'flare-log-level.warn', 'flare-log-level.error', 'flare-log-level.critical', 'flare-log-level.off'], 'order': []}, {'id': 'f6c80fdcc00b702c54ff6ae5ff2ac7f16d9afef109bdf53ee990376455301ab2', 'revision': 1715109076236, 'schema_version': 'v1.0.0', 'action': 'enable', 'lib_config': {'library_language': 'all', 'library_version': 'latest', 'service_name': 'zachs-python-app', 'env': 'zachariah', 'tracing_enabled': True, 'dynamic_sampling_enabled': False, 'tracing_tags': ['rc:works'], 'tracing_sampling_rules': [{'service': 'zachs-python-app', 'provenance': 'customer', 'resource': 'GET /', 'sample_rate': 0.01}, {'service': 'zachs-python-app', 'provenance': 'customer', 'resource': '', 'sample_rate': 1}]}, 'service_target': {'service': 'zachs-python-app', 'env': 'zachariah'}}], 'shared_data_counter': 1}
```

The python tracer’s implementation of pulling RC rules was brittle and relied upon data["config"][0] always having the lib_config dict. However, with tracer flares it seems the agent sometimes sends the payload with the flare info in that 0 position in the list, so instead we need to sometimes grab data["config"][1] .

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by juanjux — APPROVED** [review]

File: ddtrace/settings/config.py [hunk 1]
```diff
@@ -743,8 +743,14 @@ def _handle_remoteconfig(self, data, test_tracer=None):
             log.warning("unexpected number of RC payloads %r", data)
             return
 
+        # Check if 'lib_config' is a key in the dictionary since other items can be sent in the payload
+        config = None
+        for config_item in data["config"]:
+            if isinstance(config_item, Dict):
+                if "lib_config" in config_item:
+                    config = config_item
```
Comment (erikayasuda) [inline ddtrace/settings/config.py:751]: nit: Should we just `break` here when we find one?

File: tests/internal/test_settings.py [hunk 2]
```diff
@@ -182,7 +196,7 @@ def test_settings_missing_lib_config(config, monkeypatch):
     base_rc_config = _base_rc_config({})
 
     # Delete "lib_config" from the remote config
-    del base_rc_config["config"][0]["lib_config"]
+    del base_rc_config["config"][1]["lib_config"]
```
Comment (erikayasuda) [inline tests/internal/test_settings.py:199]: Do you know why we originally tested for this? 🤔

**Review by erikayasuda — APPROVED** [review]
Just a few non-blockers

File: ddtrace/settings/config.py (hunk 1 above)
Comment (ZStriker19) [reply ddtrace/settings/config.py:751]: proly

File: tests/internal/test_settings.py (hunk 2 above)
Comment (ZStriker19) [reply tests/internal/test_settings.py:199]: We're not really "testing for this" per say, I guess we just wanted to delete the lib_config, and then see what it does? In this case we're doing the same thing, it's just that lib_config lives one list item later.

File: ddtrace/settings/config.py [hunk 3]
```diff
@@ -743,8 +743,15 @@ def _handle_remoteconfig(self, data, test_tracer=None):
...
+                    break
```
Comment (gnufede) [inline ddtrace/settings/config.py:752]: nit you can join the ifs:
```suggestion
            if isinstance(config_item, Dict) and "lib_config" in config_item:
                config = config_item
                break
```

**Review by gnufede — APPROVED** [review]

**Pull request merged**